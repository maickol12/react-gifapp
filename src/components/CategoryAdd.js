import Reat, { useState } from 'react';
import PropTypes from 'prop-types';


const CategoryAdd = ({ setCategory }) => {
    const [inputValue, setInputValue] = useState('');

    const handleInputChange = (e) => {
        setInputValue(e.target.value);
    }
    const handleSubmit = (e) => {
        e.preventDefault();
        if( inputValue.trim().length > 2 ){
            setCategory( value => [inputValue, ...value]);
            setInputValue('')
        }
    }
    return(
        <div>
            <form onSubmit={handleSubmit}>
                <p>{ inputValue }</p>
                <input
                    type="text"
                    value={ inputValue }
                    onChange={ handleInputChange } />
                
            </form>
        </div>
    )
}

CategoryAdd.propTypes = {
    setCategory: PropTypes.func.isRequired
}

export default CategoryAdd;