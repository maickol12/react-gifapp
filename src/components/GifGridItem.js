import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ( {id,title,url} ) => { 
    return(
        <div className='card animate__fadeIn_10s'>
            <img src={ url } alt={ title } />
            <p>{ title }</p>
        </div>
    )
}

GifGridItem.prototype = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}