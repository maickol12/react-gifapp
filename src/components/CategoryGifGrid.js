//import React, { useState, useEffect } from 'react';
import React, { useState, useEffect } from 'react';
//import { getGifs } from '../helper/getGifs';
import { useFetchGifs } from '../hooks/useFetchGif';
import { GifGridItem } from './GifGridItem';
import PropTypes from 'prop-types';

export const CategoryGifGrid = ({ category }) => {
    // const [images, setImages] = useState([]);
    // const [count, setCount ] = useState(0);
    
    const {data,loading} = useFetchGifs(category);

    

    // useEffect( () => {
    //     getGifs(category).then( setImages );
    // } , []);

   
    return(
        <div>
            <h3>Category { category }</h3>
            { loading && <p className="animate__animated animated__flash">Cargando</p>}
            <div className='card-grid'>
                {
                    data.map((img) => (
                        <GifGridItem 
                            key={img.id+img.title}
                            {...img} />
                    ))
                }
            </div>
        </div>
    )
};

CategoryGifGrid.propTypes = {
    category: PropTypes.string.isRequired
}