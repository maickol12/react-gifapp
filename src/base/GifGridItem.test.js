import { GifGridItem } from "../components/GifGridItem";
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';

describe('Prueba <GifGridItem />',() => {
    const title = 'Un t√≠tulo';
    const url   = 'https://localhost/algo.jpg';
    const wraper = shallow( <GifGridItem title={title} url={url} /> );

    test('debe de mostrar <GifGridItem /> correctamente', () => {
        expect ( wraper ).toMatchSnapshot();
    });
    
    test('debe de tener un parrago con el title', () => {
        const p = wraper.find('p');
        
        expect( p.text().trim() ).toBe( title );
    }); 
    test('debe de tener la imagen igual al url y alt de los props',() => {
        const img = wraper.find('img');

        expect( img.prop('src') ).toBe( url );
        expect( img.prop('alt') ).toBe( title );
    });
    test('debe de tener animate__fadeIn_10s', () => {
        const div = wraper.find('div');
        const className = div.prop('className');
        expect( className.includes('animate__fadeIn_10s') ).toBe( true );
    });
});