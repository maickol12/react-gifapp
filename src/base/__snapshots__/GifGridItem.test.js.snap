// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba <GifGridItem /> debe de mostrar <GifGridItem /> correctamente 1`] = `
<div
  className="card animate__fadeIn_10s"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
